src/utils/so/Value.cc

```diff
@@ -37,9 +37,26 @@ bool drafter::utils::so::operator==(const Number& lhs, const Number& rhs)
     return lhs.data == rhs.data;
 }
 
+namespace
+{
+    bool hasEqualMembers(const Object& lhs, const Object& rhs)
+    {
+        using itemType = typename Object::container_type::const_reference;
+        for (itemType item : lhs.data) {
```

Comment:
If I understand this logic correctly, if there is the following I think they'll be treated as equal.

LHS:

```
+ name: doe
```

RHS:

```
+ name: doe
+ additional: true
```

Since the logic loops over lhs members and verifies they are equal to rhs. Additional RHS members are ignored.

Perhaps a guard that the length of lhs and rhs are equal at the top of the function would solve this problem?

Reply:
There is on upper level checked count of members is equal. 

https://github.com/apiaryio/drafter/pull/714/files#diff-c71b7d4a5846e44057ccc118bacbe934R59

So situation you describe should not happen. But if you like I can make it in one step if you like

Reply:
@klokane you're right, I overlooked that part.